server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -328,6 +345,100 @@ void addNodes(int newNodesCount) {
             }
         }
 
+        void runRandomly() {
+
+            final int randomSteps = scaledRandomIntBetween(10, 10000);
+            logger.info("--> start of safety phase of at least [{}] steps", randomSteps);
+
+            deterministicTaskQueue.setExecutionDelayVariabilityMillis(EXTREME_DELAY_VARIABILITY);
+            int step = 0;
+            long finishTime = -1;
+
+            while (finishTime == -1 || deterministicTaskQueue.getCurrentTimeMillis() <= finishTime) {
+                step++;
+                if (randomSteps <= step && finishTime == -1) {
+                    finishTime = deterministicTaskQueue.getLatestDeferredExecutionTime();
+                    deterministicTaskQueue.setExecutionDelayVariabilityMillis(DEFAULT_DELAY_VARIABILITY);
+                    logger.debug("----> [runRandomly {}] reducing delay variability and running until [{}]", step, finishTime);
```

Comment:
This is something I wanted to do in the PoC but never got round to it. We had a few failures which were to do with an unrealistically-delayed action that occurred long into the following stabilisation phase, and had to lengthen stabilisation as a result. I think this extra time should be in the safety phase instead.